streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -58,49 +71,127 @@ class WriteAheadLogSuite extends SparkFunSuite with BeforeAndAfter {
     Utils.deleteRecursively(tempDir)
   }
 
-  test("WriteAheadLogUtils - log selection and creation") {
-    val logDir = Utils.createTempDir().getAbsolutePath()
+  test(testPrefix + "read all logs") {
+    // Write data manually for testing reading through WriteAheadLog
+    val writtenData = (1 to 10).map { i =>
+      val data = generateRandomData()
+      val file = testDir + s"/log-$i-$i"
+      writeDataManually(data, file)
```

Comment:
It's very important that the BatchedWAL can recover when the existing data is not batched as well, right? I mean, isn't it possible that some users start their stream using Spark 1.5. Then with the release, they want to use the BatchedWAL, therefore they enable it. All the existing data in the checkpoint directory is not batched, but the BatchedWAL should be able to recover from it still